lib/pubmed/map_pub_hash.rb

```diff
@@ -0,0 +1,270 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+module Pubmed
+  # Map Pubmed source record data into the SUL PubHash data
+  class MapPubHash
+    attr_reader :pubmed_article
```

Comment:
this is the new class that gets all of the Pubmed pub hash related methods ported over (in the Pubmed namespace)